langchain/vectorstores/chroma.py

```diff
@@ -574,6 +574,52 @@ def from_documents(
             **kwargs,
         )
 
+    @classmethod
+    def from_embeddings(
+        cls: Type[Chroma],
+        text_embeddings: List[Tuple[str, List[float]]],
+        embedding: Optional[Embeddings] = None,
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        collection_name: str = _LANGCHAIN_DEFAULT_COLLECTION_NAME,
+        persist_directory: Optional[str] = None,
+        client_settings: Optional[Any] = None,
+        client: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> Chroma:
+        """Create a Chroma vectorstore from a list of embeddings.
+
+        If a persist_directory is specified, the collection will be persisted there.
+        Otherwise, the data will be ephemeral in-memory.
+
+        Args:
+            text_embeddings (List[Tuple[str, List[float]]]): List of tuples with texts
+            and embeddings (list of floats).
+            metadatas (Optional[List[dict]]): List of metadatas. Defaults to None.
+            collection_name (str): Name of the collection to create.
+            persist_directory (Optional[str]): Directory to persist the collection.
+            ids (Optional[List[str]]): List of document IDs. Defaults to None.
+            embedding (Optional[Embeddings]): Embedding function. Defaults to None.
```

Comment:
Should `embedding` for embedding function be optionally supplied if method requires embeddings to be provided already?

Reply:
Good point, the ``embedding`` function indeed need to embed the queries later when the vectorstore is used. I pushed a commit to fix this.